### ISSUE TITLE
Add mechanism to cryptographically secure non-credential VP properties (contexts etc)

### ISSUE THREAD

**Issue by dmitrizagidulin** [w3c/vc-data-model#1360]
(Extracting from issue #1265, as requested on the previous call.)
Verifiable Presentations need a way to cryptographically secure externally linked resources (such as `@context`s) that is similar to the VC `relatedProperties`/`digestSRI` mechanism.

**Comment by dmitrizagidulin** [comment]
My proposal - we add the `relatedProperties` field to the VP data model. (That is, extend the domain of the `relatedProperties` field to be _either_ VC or VP.)

**Comment by OR13** [comment]
this is not an issue for vc-jose-cose, additional claims can be present, and if not understood they are ignored.

in the context of rdf, they get assigned "issuer dependent terms".

**Comment by OR13** [comment]
@dmitrizagidulin that functionality is still supported by what I said... and it still needs to be interpreted consistently by issuers and verifiers to be useful.

**Comment by dmitrizagidulin** [comment]
@OR13 I agree that schema-wise, it's not an issue for jose/cose serializations. However, we should still highlight its use in VPs, _so that_ verifiers interpret it consistently

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-11-28-vcwg-special#section1-2) on 2023-11-28

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.2. Add mechanism to cryptographically secure non-credential VP properties (contexts etc) (issue vc-data-model#1360)

_See github issue [vc-data-model#1360](https://github.com/w3c/vc-data-model/issues/1360)._




**Dmitri Zagidulin:** We want to be able to lock these down cryptographically.  
… We have only added this for verifiable credentials.  
… But not for related resources.  
… We need to do this for VPs.  

**Brent Zundel:** Can the solution be as simple as saying "this section also refers to VPs".  

**Manu Sporny:** Do you mean "related resources"?  
… I can volunteer to write a PR for this.  

> *Joe Andrieu:* correct. id we need to secure contexts we should do that. arbitrary resources is a layer violation.

> *Phillip Long:* pdl-asu has joined #vcwg-special.

> *Joe Andrieu:* happy to review pr.

</details>

**Comment by msporny** [comment]
PR #1370 has been raised to address this issue. This issue will be closed once issue #1370 has been merged.

**Comment by msporny** [comment]
PR #1370 has failed to gain consensus.

**Comment by dmrazzy** [comment]
Workflow
